mlflow/langchain/__init__.py

```diff
@@ -328,6 +345,27 @@ def log_model(
             type(lc_model.agent.llm_chain.llm).__name__,
         )
 
+    if isinstance(lc_model, langchain.chains.RetrievalQA):
+        if conda_env is None and pip_requirements is None:
+            logger.warning(
+                "RetrievalQA models are currently incompatible with the "
+                "infer_pip_requirements function. Please specify the environment "
+                "requirements in the `conda_env` or `pip_requirements` parameter."
+            )
+
+        loader_fn = None
+        persist_dir = None
+        if metadata is not None:
+            if LOADER_FN_KEY in metadata:
+                loader_fn = metadata.pop(LOADER_FN_KEY)
+            if PERSIST_DIR_KEY in metadata:
+                persist_dir = metadata.pop(PERSIST_DIR_KEY)
+        mlflow.pyfunc.log_model(
```

Comment:
IMO these should be part of the LangChain flavor config in the MLmodel file, and we should introduce top-level arguments in `mlflow.langchain.log_model()` / `mlflow.langchain.save_model()` that users can specify for this use case. These were included in your design, right?

Reply:
I'm assuming that we may prefer to avoid adding top-level arguments in the implementation. I can add them.